horovod/run/util/network.py

```diff
@@ -44,7 +44,20 @@ def resolve_host_name(host_name):
 
 
 # Given server factory, find a usable port
-def find_port(server_factory):
+def find_port(server_factory, pedl_provisioned_port=0, verbose=False):
+    if verbose:
+        print(f"Find port pedl_provisioned_port: {pedl_provisioned_port}.")
+    if pedl_provisioned_port > 0:
+        try:
+            addr = ('', pedl_provisioned_port)
+            server = server_factory(addr)
+            if verbose:
+                print(f"Find port connected to {pedl_provisioned_port}.")
+            return server, pedl_provisioned_port
+        except Exception as e:
+            pass
+        raise Exception(f"Unable to bind to pedl_provisioned_port {pedl_provisioned_port}.")
```

Comment:
```        
        except Exception as e:
            pass
        raise Exception(f"Unable to bind to pedl_provisioned_port {pedl_provisioned_port}.")
```

should ideally instead be something like

```
        except SomeSpecificException as e:
            raise Exception(f"Unable to bind to pedl_provisioned_port {pedl_provisioned_port}.")
```
and maybe even there's some `Horovod` specific type of error or PEDL error we can raise. You can opt to do this later like the other change I mentioned earlier (but they're both pretty quick all things considered)

Reply:
Hmm, this is a good point. I am not sure what `SomeSpecificException` in this case is and just followed the existing code style here in Horovod (around L73).